### ISSUE TITLE
[macOS] "kitty needs to be updated" warning message

### ISSUE THREAD

**Issue by screendriver** [kovidgoyal/kitty#4717]
**Is your feature request related to a problem? Please describe.**

On macOS Monterey (12.2) I'm getting the message

```
"kitty" needs to be updated


This app will not work with future versions of macOS and needs to be updated to improve compatibility.
Contact the developer for more information.
```

<img width="262" alt="Screenshot 2022-02-17 at 22 24 58" src="https://user-images.githubusercontent.com/149248/154638132-14c9515a-41e8-4d24-a5d6-374f2146c594.png">

Is that something that needs to be handled or can we just ignore that message?

**Comment by kovidgoyal** [comment]
I dont get any such message running kitty on macOS 12.2.1. Are you
running an up to date version of kitty? And what does click Learn more
tell you?

**Comment by screendriver** [comment]
> That's because you're running macOS's own deprecated cli program inside kitty

What do you mean with "cli program"?

> a program that calls deprecated API

I'm not calling anything in `kitty`. I'm getting this message when I try to open `kitty`. A message that I don't get with `iTerm`, `Alacritty` and so on 🤔 

> Are you running an up to date version of kitty?

Yupp. `0.24.2`

> And what does click Learn more tell you?

I don't get this message anymore and already clicked on `Ok`. I was updating to macOS `12.2.1` and after a reboot I received that message. The same when I updated to `12.2` previously. The next time when I get this message again I will click on `Learn more` and paste a screenshot here. Thanks for the quick reply.

**Comment by page-down** [comment]
> What do you mean with "cli program"?

Command Line Programs

This includes programs that are called in shell rc. As well as certain functions such as autocomplete that automatically try to run and check the cli program version.

E.g. `/usr/bin/python`
`~/Library/PythonWrapper/promptedapps.plist`

```xml
<dict>
  <key>/Applications/iTerm.app/Contents/MacOS/iTerm2</key>
  <string>Version 12.2.1 ...</string>
  <key>/Applications/kitty.app/Contents/MacOS/kitty</key>
  <string>Version 12.2.1 ...</string>
</dict>
```

Click `Learn more`
https://www.python.org/doc/sunset-python-2/

> ... I was updating to macOS 12.2.1 and after a reboot I received that message. The same when I updated to 12.2 previously. ...

I also encounter this after every update of macOS 12.

**Comment by screendriver** [comment]
Ah, now I got the message again on another macOS machine and I clicked on `Learn more` @kovidgoyal and it opened, as @page-down already mentioned https://www.python.org/doc/sunset-python-2/

I'm asking myself what is using Python on my machine (except `kitty` itself) because most of the tools that I use are written in `Rust` or `Node.js` 🤔

**Comment by screendriver** [comment]
Ok. I will try to set an alias to my Hombrew Python3 installation like `alias python="/opt/homebrew/bin/python3"` and see if that helps.

**Comment by page-down** [comment]
> I also encounter this after every update of macOS 12.

I found out why this warning is prompted in my environment.
Neovim will try to run `python` when opening a py file, like checking the version.
Even though I created a symbolic link in the PATH, it still runs `python` from `/usr/bin`.

https://github.com/neovim/neovim/blob/v0.6.1/runtime/autoload/health/provider.vim

E.g.
```shell
python -c "import platform, sys; sys.stdout.write(platform.python_version())"
```

> ... most of the tools that I use are written in Rust or Node.js ...

This has nothing to do with the language in which the tool is implemented.
As far as I know many prompts, such as `starship`, will run `python --version` to check the version and show it.

Because of the special nature of the terminal emulator and Apple's ambiguous alert, it leads to the misconception that this is a problem with kitty.
